src/util/attributesComparator.ts

```diff
@@ -0,0 +1,26 @@
+import { ElementAst, BoundElementProperty, AttrAst } from '@angular/compiler';
+import { getAttributeValue } from './getAttributeValue';
+import { getLiteralValue } from './getLiteralValue';
+
+function attributesComparator(baseAttributes: any = [], el: ElementAst): boolean {
+  const attributes: Array<AttrAst | BoundElementProperty> = [].concat(...(el.attrs as any), ...(el.inputs as any));
```

Comment:
I'd suggest to refactor it to use sets/maps, so we can drop the quadratic check to a linear one.

Reply:
Also, if possible, can we remove these `any` casts?

Reply:
@rafaelss95 done

@mgechev I have two different arrays, I don't think I can escape the quadratic check here